shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1253,60 +1262,72 @@ - (FlutterTextInputView*)getOrCreateAutofillableView:(NSDictionary*)field
     inputView =
         needsPasswordAutofill ? [FlutterSecureTextInputView alloc] : [FlutterTextInputView alloc];
     inputView = [[inputView init] autorelease];
-    [self addToKeyWindowIfNeeded:inputView];
+    [self addToInputParentViewIfNeeded:inputView];
   }
 
   inputView.textInputDelegate = _textInputDelegate;
   [inputView configureWithDictionary:field];
   return inputView;
 }
 
-// Removes every installed input field, unless it's in the current autofill
-// context. May remove the active view too if includeActiveView is YES.
-// When clearText is YES, the text on the input fields will be set to empty before
-// they are removed from the view hierarchy, to avoid autofill save .
-- (void)cleanUpViewHierarchy:(BOOL)includeActiveView clearText:(BOOL)clearText {
+// The UIView to add FlutterTextInputViews to.
+- (UIView*)textInputParentView {
   UIWindow* keyWindow = [UIApplication sharedApplication].keyWindow;
   NSAssert(keyWindow != nullptr,
            @"The application must have a key window since the keyboard client "
            @"must be part of the responder chain to function");
+  return keyWindow;
+}
 
-  for (UIView* view in keyWindow.subviews) {
+// Removes every installed input field, unless it's in the current autofill
+// context. May remove the active view too if includeActiveView is YES.
+// When clearText is YES, the text on the input fields will be set to empty before
+// they are removed from the view hierarchy, to avoid triggering autofill save.
+- (void)cleanUpViewHierarchy:(BOOL)includeActiveView clearText:(BOOL)clearText {
+  for (UIView* view in self.textInputParentView.subviews) {
     if ([view isKindOfClass:[FlutterTextInputView class]] &&
         (includeActiveView || view != _activeView)) {
       FlutterTextInputView* inputView = (FlutterTextInputView*)view;
       if (_autofillContext[inputView.autofillId] != view) {
         if (clearText) {
-          inputView.text.string = @"";
+          [inputView replaceRangeLocal:NSMakeRange(0, inputView.text.length) withText:@""];
         }
         [view removeFromSuperview];
       }
     }
   }
 }
 
-- (void)changeInputViewsAutofillVisibility:(BOOL)newVisibility {
-  UIWindow* keyWindow = [UIApplication sharedApplication].keyWindow;
-  NSAssert(keyWindow != nullptr,
-           @"The application must have a key window since the keyboard client "
-           @"must be part of the responder chain to function");
+- (void)collectGarbageInputViews {
+  [self cleanUpViewHierarchy:NO clearText:YES];
```

Comment:
We don't know what the context is anymore after the async. You should check whether dealloc'ed this plugin etc.

Reply:
Does `performSelector` not retain `self`?

Reply:
Or you mean there could be multiple instances of this plugin, adding/removing on the keywindow could potentially mess up other text input plugin instances?